### PR TITLE
 Registry should not be deployed when Azure SP is used

### PR DESCRIPTION
**Pull request by savitharaghunathan**
Registry should not be deployed when Azure SP is used

### CODE REVIEW

File: api/v1alpha1/oadp_types.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ const ReconciledReasonError = "Error"
 const ReconcileCompleteMessage = "Reconcile complete"
 
 const OadpOperatorLabel = "openshift.io/oadp"
+const RegistryDeploymentLabel = "openshift.io/registryDeployed"
```
Comment (dymurray) [inline api/v1alpha1/oadp_types.go:32]: NIT: `I would do openshift.io/oadp-registry`

File: controllers/bsl.go [hunk 2]
```diff
@@ -151,14 +151,20 @@ func (r *DPAReconciler) updateBSLFromSpec(bsl *velerov1.BackupStorageLocation, d
 	if err != nil {
 		return err
 	}
-
+	registryDeployment := "True"
+	if bslSpec.Provider == "azure" {
+		if len(bslSpec.Config["storageAccountKeyEnvVar"]) == 0 {
```
Comment (dymurray) [inline controllers/bsl.go:156]: This looks great, please just add a comment here explaining why we are doing this

File: api/v1alpha1/oadp_types.go (hunk 1 above)
Comment (savitharaghunathan) [reply api/v1alpha1/oadp_types.go]: resolved. Thanks, Dylan :)

File: controllers/bsl.go (hunk 2 above)
Comment (savitharaghunathan) [reply controllers/bsl.go:159]: Added a blurb. Let me know if that makes sense or not

**Review by dymurray — APPROVED** [review]

**Pull request merged**